guilib/measure/classes/abc.py

```diff
@@ -0,0 +1,603 @@
+        config : ConfigParser
+            The settings to check.
+        match_exactly : bool
+            Whether obj_info should be matched exactly.
+
+        Returns
+        -------
+        sorting_info : tuple
+            A tuple that can be used to sort the detected settings.
+            Larger values in the tuple indicate a higher degree of
+            conformity. The order of the items in the tuple is the
+            order of their significance. This return value is used
+            to determine the best-matching settings files when
+            multiple files are found.
+        """
+
+    @classmethod
+    @abstractmethod
+    def is_matching_user_settings(cls, obj_info, config, match_exactly):
+        """Determine if a `config` file is for this instance.
+
+        Subclasses must extend this method by calling
+        super().is_matching_user_settings at the beginning.
+
+        This method must use obj_info to determine if a config is
+        suitable for the instance. To perform this check one can
+        .get values from config and compare the information from
+        obj_info.more and obj_info.unique_name to it. If
+        match_exactly is True, the config has to be a perfect match
+        and must allow full functionality of the connected hardware.
+
+        Parameters
+        ----------
+        obj_info : SettingsInfo
+            The information that should be used to check `config`.
+        config : ConfigParser
+            The settings to check.
+        match_exactly : bool
+            Whether obj_info should be matched exactly.
+
+        Returns
+        -------
+        sorting_info : tuple
+            A tuple that can be used to sort the detected settings.
+            Larger values in the tuple indicate a higher degree of
+            conformity. The order of the items in the tuple is the
+            order of their significance. This return value is used
+            to determine the best-matching settings files when
+            multiple files are found.
+        """
+        if not isinstance(obj_info, SettingsInfo):
+            raise TypeError(
+                f'obj_info should be of type SettingsInfo '
+                f'but is of type {type(obj_info).__name__}.'
+                )
+        return tuple()
+
+    @classmethod
+    @abstractmethod
+    def is_settings_for_this_class(cls, config):
+        """Determine if a `config` file is for this class.
+
+        This method must use cls attributes to determine if
+        a config is suitable for the class.
+
+        Parameters
+        ----------
+        config : ConfigParser
+            The settings to check.
+
+        Returns
+        -------
+        is_suitable : bool
+            True if the settings file is for the class.
+        """
+
+    def are_settings_invalid(self, new_settings):
+        """Check if there are any invalid settings.
+
+        Reimplementations can add additional mandatory settings at
```

Comment:
```suggestion
        Subclasses may add additional mandatory settings at
```

Reply:
Perhaps an indication on how to do that would be useful. Either here or (better) in the description of the `_mandatory_settings` attribute above.